src/os_posix/os_thread.c

```diff
@@ -67,10 +67,32 @@ __wt_thread_join(WT_SESSION_IMPL *session, wt_thread_t tid)
 
 /*
  * __wt_thread_id --
+ *	Return an arithmetic representation of a thread ID on POSIX.
+ */
+void
+__wt_thread_id(size_t *id)
+    WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
+{
+	pthread_t self;
+
+	/*
+	 * POSIX 1003.1 allows pthread_t to be an opaque type; on systems where
+	 * it's a pointer, print the pointer to match gdb output.
+	 */
+	self = pthread_self();
+#ifdef __sun
+	*id = (size_t)self;
+#else
+	*id = (size_t)(void *)self;
+#endif
+}
+
+/*
+ * __wt_thread_descriptor --
  *	Fill in a printable version of the process and thread IDs.
  */
 int
-__wt_thread_id(char *buf, size_t buflen)
+__wt_thread_descriptor(char *buf, size_t buflen)
```

Comment:
Very minor nit - I'd prefer `__wt_thread_str` because to me the word `descriptor` makes me think of a resource like a file descriptor that we need to keep track of or release, etc.